iModelCore/GeomLibs/geom/test/PolyfaceTest/t_polyface.cpp

```diff
@@ -4604,6 +4605,126 @@ TEST (PolyfaceQuery, FacetOrientation)
     Check::ClearGeometry("PolyfaceQuery.FacetOrientation");
     }
 
+// lexicographical order, with slop for equality
+static bool lexicalXYZLessThanTol(double x0, double y0, double z0, double x1, double y1, double z1, double tol = DoubleOps::SmallMetricDistance())
+    {
+    if (DoubleOps::WithinTolerance(x0, x1, tol) && DoubleOps::WithinTolerance(y0, y1, tol) && DoubleOps::WithinTolerance(z0, z1, tol))
+        return false;
+    if (!DoubleOps::WithinTolerance(x0, x1, tol))
+        {
+        if (x0 < x1)
+            return true;
+        if (x0 > x1)
+            return false;
+        }
+    if (!DoubleOps::WithinTolerance(y0, y1, tol))
+        {
+        if (y0 < y1)
+            return true;
+        if (y0 > y1)
+            return false;
+        }
+    if (!DoubleOps::WithinTolerance(z0, z1, tol))
+        {
+        if (z0 < z1)
+            return true;
+        if (z0 > z1)
+            return false;
+        }
```

Comment:
```suggestion
    if (!DoubleOps::WithinTolerance(x0, x1, tol))
        return x0 < x1;
    if (!DoubleOps::WithinTolerance(y0, y1, tol))
       return y0 < y1;
    if (!DoubleOps::WithinTolerance(z0, z1, tol))
        return z0 < z1;
```

Reply:
Also what do you return if x0 == x1?

Reply:
> Also what do you return if x0 == x1?

false. But that's a good simplification. Thanks!